### PR TITLE
To do list icons toggle

### PR DESCRIPTION
**Pull request by qho-queenie**
Toggling the arrow icons depending on the currently sorted column, and whether its ascending or decsending.

1. If its currently sorted ascendingly, the up icon shows.
2. if this is not the currently sorted column, both of the up and down arrow shows.
Screen Shot 2021-04-13 at 9 26 48 PM

3. If it is currently sorted descendingly, the down icon shows.
Screen Shot 2021-04-14 at 11 22 05 AM

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -73,8 +68,12 @@ const App = () => {
...
         let direction;
 
-        // 1. if exists -->> flip it
-        // 2. if dont exists --> destroy everything in it, create this key
+        // 1. if dont exists --> destroy everything in it, create this key
+        // 2. if exists -->> flip it
+
+        if (sortConfig['columnKey'] !== columnToSort) {
+            direction = 'asc';
+        }
```
Comment (qho-queenie) [inline src/App.js:76]: This if is needed if it has never been sorted

File: src/App.js [hunk 2]
```diff
@@ -93,7 +92,7 @@ const App = () => {
             let direction = sortConfig['dirToSort'];
 
             if (todos) {
-                if (colToSort === 'date' && direction === 'asc') {
+                if (colToSort === 'date' && direction === 'des') {
```
Comment (qho-queenie) [inline src/App.js:95]: the original logic was wrong

File: src/SortableTableHeader.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        )
+    } else {
+        return (
+            <button type='button' onClick={() => onSortColumn(colName)}>
+                {colName}
+                {sortConfig['dirToSort'] === 'asc' ?
+                    <FontAwesomeIcon icon={faSortDown} /> :
+                    <FontAwesomeIcon icon={faSortUp} />
+                }
+            </button>
```
Comment (qho-queenie) [inline src/SortableTableHeader.js]: I am using an if else to make this more readable, instead of having 2 ternaries

File: src/App.js (hunk 1 above)
Comment (kevinhongw) [reply src/App.js:76]: Why not just declare it when you init the component?

File: src/SortableTableHeader.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        )
+    }
```
Comment (kevinhongw) [inline src/SortableTableHeader.js]: Seems like the only difference is the icon name.
I would move the logic outside of rendering logic.


```suggestion
    let icon = faSort;
    if (sortConfig['columnKey'] === colName) {
      icon = sortConfig['dirToSort'] === 'asc' ? faSortDown : faSortUp
    }
    return (
        <button type='button' onClick={() => onSortColumn(colName)}>
            {colName}
            <FontAwesomeIcon icon={icon} />
        </button>
    )
```

**Review by kevinhongw — CHANGES_REQUESTED** [review]
So sorry for the late review. I missed the notification and didn't see it until today.
Good job on creating a new component! Syntax looks good. Just optimizing your logic.

**Pull request merged**